### PR TITLE
feat(undo): add vundo

### PR DESCRIPTION
**Pull request by danilevy1212**
Added vundo recipe for undo-fu users, so they can't stop being jealous of undo-trees visualizer! Changes are gated behind opting out of the `+tree` flag in the `:emacs undo` module.

### CODE REVIEW

**Comment by danilevy1212** [comment]
I feel like there should be a mention of this package on the module `readme.org`. Happy to add theme to rewrite-docs if need be and this PR gets approved :)

**Comment by elken** [comment]
FYI Vundo is now [on ELPA](https://elpa.gnu.org/packages/vundo.html)

**Comment by danilevy1212** [comment]
@elken Thank you, updated the recipe :+1:

**Comment by hpfr** [comment]
For when Henrik gets around to this, my vote would be to put `vundo` on `SPC b u` and move `doom/sudo-save-buffer` to `SPC b U`. It would be a breaking binding change, but the replacement is read-only unless the user presses more keys in quick succession, which isn’t that likely if they were assuming the binding would save the buffer. I think it’s worth considering because `vundo` is likely to be much more commonly used than saving the buffer as root.

**Comment by hpfr** [comment]
[evil-collection now supports vundo.](https://github.com/emacs-evil/evil-collection/commit/535ad8dc52137f502bfa584925705583209aa093)

@danilevy1212 It looks like you can remove all the remaps except `doom/escape`.

**Comment by hpfr** [comment]
If you like, you can checkout this branch, download the attached, and run `git am 0001-tweak-undo-remove-bindings-in-evil-collection.txt` (or `M-x magit-am-apply-patches` :wink:) to apply a commit with the fix, and then push.

[0001-tweak-undo-remove-bindings-in-evil-collection.txt](https://github.com/doomemacs/doomemacs/files/8929846/0001-tweak-undo-remove-bindings-in-evil-collection.txt)

**Comment by danilevy1212** [comment]
Thanks @hpfr, did the changes :)

**Comment by danilevy1212** [comment]
@hpfr I added an `+EMACS28` flag detector to prevent problems from emacs 27 users

File: modules/emacs/undo/config.el [hunk 1]
```diff
@@ -51,6 +51,18 @@
         filename))))
 
 
+(use-package! vundo
+  :unless (featurep! +tree)
+  :custom
+  (vundo-glyph-alist     vundo-unicode-symbols)
```
Comment (tecosaur) [inline modules/emacs/undo/config.el:57]: Minor style nit, I don't think the extra space does any good here.

Comment (danilevy1212) [reply modules/emacs/undo/config.el]: Removed the space :)

Comment (tecosaur) [reply modules/emacs/undo/config.el]: Thanks :)

**Comment by BenediktBroich** [comment]
LGTM!

**Comment by hlissner** [comment]
Sorry for taking so long to get to this! I've merged this in 9350b44 after some modifications, and included a rudimentary keybind. Thanks for your help!

> For when Henrik gets around to this, my vote would be to put vundo on SPC b u and move doom/sudo-save-buffer to SPC b U. 

I agree, but will temporarily shelve it until I'm done moving modules out into their [own repo](https://github.com/doomemacs/modules).

**Comment by danilevy1212** [comment]
Thank you, Henrik!

**Pull request closed without merging**